# Airports

## Location & Weather 

### Task:

1. Use the information in the .CSV files found at https://github.com/hadley/nycflights13/tree/master/data-raw for your source data.
2. Write your code using Python and pandas, and deliver it in a Jupyter notebook. You should post your notebook in a GitHub repository, and send me a link to the notebook.
3. For each of the three questions listed above, you’ll need to verify your answers (in two cases by searching for additional confirming information, and in the third case being alert for possible outliers). For the first two questions above, you may want to start by generating a list of the five northernmost or easternmost U.S. airports.
4. In addition to providing code, you should state your answers to each of the above questions (in Markdown) in the Jupyter notebook.
5. To receive full credit, you’ll need to include in your Jupyter notebook both your code and your documented assumptions and any changes that you made to the data based on your research (using Markdown) in your Jupyter notebook.

#### What is the northernmost airport in the United States?

In [1]:
import pandas as pd
northern_airport = pd.read_csv('https://raw.githubusercontent.com/maddogdatascience/DATA607/master/airports.csv')
northern_airport.head()
northern_airport.sort_values("lat", ascending = False).head(5)

faa                                        name        lat         lon  \
393  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
217  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
98   AIN                          Wainwright Airport  70.638056 -159.994722   
674  K03                               Wainwright As  70.613378 -159.860350   
140  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst  
393  149  -5   A  
217   44  -9   A  
98    41  -9   A  
674   35  -9   A  
140   96  -9   A

Since the longitude for Dillant Hopkins Airport seems off compared to the other values, it is important to research at least two other sources to confirm that Dillant Hopkins Airport is indeed the northernmost airport. In order to do this, I searched on Google and found two websites that gave the correct latititude and longitude values. In my findings, it was confirmed that the latitudinal value of 72.270833 is actually -72.270833. To confirm the other values, I did the same on my own and the latitudinal values were the same. Here are the two websites in which I confirmed that Dillant Hopkins Airport is the northernmost airport in the United States: http://latitude.to/articles-by-country/us/united-states/179120/dillanthopkins-airport; http://www.airnav.com/airport/keen

##### Answer: 

Dillant Hopkins Airport

#### What is the easternmost airport in the United States?

In [3]:
easternmost = pd.read_csv('https://raw.githubusercontent.com/maddogdatascience/DATA607/master/airports.csv')
eastern_airport = easternmost[(easternmost['lat'] > 44) & (easternmost['lat'] <= 50) & (easternmost['lon'] >= -75) & (easternmost['lon'] <= -65)]
eastern_airport.head()
eastern_airport.sort_values('lon', ascending=False).head(5)

faa                                 name        lat        lon  alt  tz  \
419   EPM           Eastport Municipal Airport  44.910111 -67.012694   45  -5   
591   HUL                         Houlton Intl  46.123083 -67.792056  489  -4   
246   CAR                         Caribou Muni  46.871500 -68.017917  626  -4   
1052  PQI  Northern Maine Rgnl At Presque Isle  46.688958 -68.044797  534  -4   
1340  WFK  Northern Aroostook Regional Airport  47.285556 -68.312778  988  -5   

     dst  
419    A  
591    A  
246    A  
1052   A  
1340   A

Since the latitude and longitudinal values seemed accurate, I double-checked each value using www.airnav.com and www.latitude.to
Each value matched the values in the table above. The longitudinal value is the value that is sorted since it pinpoints west and east direction. Specifically, Eastport Municipal Airport was confirmed as the easternmost airport in the United States by these links:  http://www.airnav.com/airport/KEPM; http://latitude.to/articles-by-country/us/united-states/209992/eastport-municipal-airport

##### Answer:

Eastern Municipal Airport

#### On February 12th, 2013, which New York area airport had the windiest weather?

In [5]:
windiest = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv')
windiest_weather = windiest[(windiest["month"]==2) & (windiest["day"]==12) & (windiest["year"]==2013)]
windiest_weather.head()
windiest_weather.sort_values("wind_speed", ascending=False).head(5)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1015     EWR  2013      2   12     8  39.02  26.96  61.63     260.0   
18433    LGA  2013      2   12     7  42.98  26.06  50.94     290.0   
1024     EWR  2013      2   12    17  44.06  26.06  48.87     270.0   
18444    LGA  2013      2   12    18  44.06  23.00  43.02     300.0   
9728     JFK  2013      2   12    13  42.08  24.08  48.56     280.0   

       wind_speed    wind_gust  precip  pressure  visib             time_hour  
1015   1048.36058  1206.432388     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18433    23.01560    26.485892     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1024     21.86482    25.161598     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18444    21.86482    25.161598     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
9728     20.71404    23.837303     0.0    1011.6   10.0  2013-02-12T13:00:00Z

According to the table, EWR, or Newark Liberty National Airport, has a wind speed of 1048.36058 miles per hour. Compared to the other values that around 20 to 23 miles per hour, I believe we need to confirm the wind speed for EWR. According to Weather Underground, the max wind speed recorded for Newark was 25 miles per hour on February 12th, 2013. Checking other airports on Weather Underground, these wind speeds approximately correct by looking at data presented (i.e. tables, graphs, max wind speeds). Since the value of 1048.36058 seems like an outlier, the windiest airport may be LaGuardia Airport instead of Liberty Airport with the information given.

##### Answer:

### Without Outlier
LaGuardia Airport

### With Outlier
Newark Liberty National Airport